src/components/wrappers/PopUp.vue

```diff
@@ -0,0 +1,131 @@
+<script>
+import ButtonSign from '@/components/formElements/ButtonSign.vue';
+
+export default {
+  components: {
+    ButtonSign,
+  },
+  props: {
+    informational: Boolean,
+    stopPropagation: {
+      type: Boolean,
+      default: false,
+    },
+    buttonStyleType: {
+      type: String,
+      default: 'hint',
+    },
+    position: {
+      type: String,
+      default: 'right',
+    },
+  },
+  emits: ['close'],
+  data() {
+    return {
+      isShown: false,
+    };
+  },
+  methods: {
+    validatePopUpClosing(event) {
+      const target = document.querySelector('.pop-up-content');
+
+      if (event.target !== target && !target.contains(event.target)) {
+        this.closePopUp();
+      }
+    },
+    openPopUp() {
+      this.isShown = true;
+      document.addEventListener('click', this.validatePopUpClosing);
+    },
+    closePopUp() {
+      this.isShown = false;
+      document.removeEventListener('click', this.validatePopUpClosing);
+    },
+    togglePopUp() {
+      this[this.isShown ? 'closePopUp' : 'openPopUp']();
+    },
+  },
+};
+</script>
+
+<template>
+  <div class="pop-up">
+    <ButtonSign
+      class="toggling-button"
+      :style-type="buttonStyleType"
+      :stop-propagation="stopPropagation"
+      @click="togglePopUp"
+    />
+    <Transition>
+      <div
+        v-if="isShown"
+        class="pop-up-content"
```

Comment:
popup is one word

Reply:
https://www.collinsdictionary.com/dictionary/english/pop-up

Reply:
https://www.scribbr.com/language-rules/hyphens/
Phrasal verbs don’t normally require hyphens. However, when they are used as nouns, they either take a hyphen or are combined into a single word. If you’re not sure whether a hyphen is necessary, consult a dictionary.

checking dictionary...
https://www.collinsdictionary.com/dictionary/english/pop-up

also, usually a letter after hyphen is capitalized in abbreviations

Reply:
The main thought is it's that that combination of word create another word with another sense.
Popup or pop-up could be separated for pop and up because it's already part of this word.

And another argument - could you imagine that in future html specification there could be new tag <popup>
if you could it means that we shouldn't use it like that here